### PR TITLE
[CUDA] Add test for pcl::gpu::NormalEstimation (issue #2371, GPU NaNs)

### PR DESCRIPTION
**Pull request by ruslo**

### CODE REVIEW

File: gpu/features/test/test_normals.cpp [hunk 1]
```diff
@@ -394,6 +394,43 @@ TEST(PCL_FeaturesGPU, issue_2371)
     ne_gpu.compute(normals_gpu);
 }
 
+// See:
+// - https://github.com/PointCloudLibrary/pcl/pull/3627#discussion_r375826172
+TEST(PCL_FeaturesGPU, normals_nan_gpu)
+{
+    const std::size_t N = 5;
+
+    PointCloud<PointXYZ>::Ptr cloud(new PointCloud<PointXYZ>());
```
Comment (SergioRAgostinho) [inline gpu/features/test/test_normals.cpp]: This does not need to be a shared pointer.

File: gpu/features/test/test_normals.cpp [hunk 2]
```diff
@@ -394,6 +394,43 @@ TEST(PCL_FeaturesGPU, issue_2371)
...
+    ne_device.compute(normals_device);
+
+    std::vector<PointXYZ> downloaded;
+    normals_device.download(downloaded);
+
+    ASSERT_EQ(downloaded.size(), N);
+
+    for(std::size_t i = 0; i < downloaded.size(); ++i)
+    {
+        const PointXYZ& n = downloaded[i];
```
Comment (SergioRAgostinho) [inline gpu/features/test/test_normals.cpp]: Use a for-range loop here.
```cpp
for(const auto& n: downloaded)
{
  ...
}
```

File: gpu/features/test/test_normals.cpp (hunk 1 above)
Comment (ruslo) [reply gpu/features/test/test_normals.cpp]: Fixed

File: gpu/features/test/test_normals.cpp (hunk 2 above)
Comment (ruslo) [reply gpu/features/test/test_normals.cpp]: Fixed

**Pull request merged**